apps/authentication/api/connection_token.py

```diff
@@ -675,3 +678,10 @@ def check_permissions(self, request):
 
     def get_queryset(self):
         return AdminConnectionToken.objects.all()
+
+    def get_permed_account(self, user, asset, account_name, protocol):
+        with tmp_to_org(asset.org):
+            account = asset.accounts.all().active().get(name=account_name)
+            account.actions = ActionChoices.all()
+            account.date_expired = timezone.now() + timezone.timedelta(days=365)
+            return account
```

Comment:
This is already an up-to-date version of the given code snippet as per 2021-09-01 and doesn't contain any irregularities, potential issues, or optimizations that need to be addressed.